src/metrics/include/metrics/metrics.hpp

```diff
@@ -89,6 +92,25 @@ class RegistryT final {
         return OperationT{opIt->second};
     }
 
+    OperationT<ClockSource> operation(std::string actorName,
+                                      std::string opName,
+                                      ActorId actorId,
+                                      genny::TimeSpec threshold,
+                                      double_t percentage) {
+        auto& opsByType = this->_ops[actorName];
+        auto& opsByThread = opsByType[opName];
+        auto opIt =
+            opsByThread
+                .try_emplace(
+                    actorId,
+                    std::move(actorName),
+                    std::move(opName),
+                    std::make_optional<typename OperationImpl<ClockSource>::OperationThreshold>(
+                        threshold, percentage))
+                .first;
+        return OperationT{opIt->second};
+    }
```

Comment:
1. this is a little copy/pasta from above is there a way to DRY it up?
2. the spacing looks almost a  bit lisp-y is this really how clang-format leaves it?

Reply:
Agree with both points, although I'm not sure if you can DRY it up cleanly. Would be nice if you could either have one method with default params or use method chaining, but that might require having making the new params be a single `std::Optional<genny::TimeSpec, double_t>` which would clash with your  `std::optional<OperationThreshold>` type. Seems like it might be most legible to keep it as it is unless there's some other c++ voodoo you know of?